### DISCUSSION TITLE
Home Maintenance Tracking

### DISCUSSION THREAD

**Discussion by varunpan** [sysadminsmedia/homebox#10] [Ideas]
As a new Home Owner, I want to be able to track my home maintenance tasks like cleaning AC unit, Changing Filters, Cleaning Deck, Water heater tank maintenance. There are a ton of tasks that a home owner needs to keep track of and hoping to make homebox the home of tracking these tasks. 

I already track all my appliances and equipment with manuals/receipts in homebox, but it is missing the maintenance component so that I can show that I have been taking care of it and easily be able to pull up the records when needed.

Please let me know your thoughts on this!

**Comment by tankerkiller125** [comment]
I think that this would fit into the project pretty well honestly, it would be more of a long term goal at the moment (like several months off) simply due to needing to learn the codebase, get things up and running, etc. but it's for sure something I think would be good to implement.

Thoughts on this one @katosdev?

**Reply by katosdev** [reply]
Pretty sure that maintenance logs are already present, and that the functionality that you are after is available now.
If there anything fundamentally missing that you’d like to see @varunpan ?

I’ll do some testing and see what maintenance features might need building out. Being honest it’s not a feature that I’ve much utilised personally, but definitely sounds like a useful one!

CC @tankerkiller125

**Reply by varunpan** [reply]
Agreed, I love using this, but it is missing some components like:

- Make a recurring maintenance item, for example, I want to be able to create a task to change the Furnace filter every month and have it generate a maintenance item for it (preferable a cal event). 
- The ability to tag repairs vs maintenance/service tasks.
- The ability to load file/receipts related to maintenance tasks
- consume parts used in maintenance - for examples, I have item of 12x furnace filters, I want to 'consume' a filter when I complete that maintenance task. 
- See Overdue/upcoming maintenance items on the home page.

These are items I can think of from the top of my head.

**Reply by katosdev** [reply]
Ahh these are some very fair suggestions! I think adding a "maintenance type" for repair / service / part replacement etc would be a good move. Thoughts @tankerkiller125 ?

**Reply by tankerkiller125** [reply]
I think that all of these are fair ideas, starting with maintenance type would be a first implementation step.

**Reply by jjmoffitt** [reply]
There was an issue open for this in the original that may have some more details that could help. https://github.com/hay-kot/homebox/issues/516#issuecomment-2120829202

This is something I'm very excited could be possible. As of now I'm trying to cludge this together using task management software and it would be so nice to see it attached to items.

**Reply by 101br03k** [reply]
Someone made [homebox/issues/172](https://github.com/sysadminsmedia/homebox/issues/172) and I put a Bounty of 15 on it.

**Reply by temmaywatson-dot** [reply]
Love this idea for Homebox! As a homeowner, keeping up with tasks like cleaning the AC unit, changing filters, or maintaining the water heater is a hassle. Adding a maintenance tracking feature to Homebox, where I can log these tasks and store records alongside my appliance manuals and receipts, would be amazing. It’d also help prove I’ve maintained my appliances for warranty claims or repairs. Maybe include reminders for routine tasks like those we see at [Appliance Repair](https://appliancerepairstocktonca.com/) or a maintenance log tied to each appliance? Excited for this—please share updates!

**Comment by Jackxwb** [comment]
Regarding regular maintenance, I have come up with several models
Depending on the type of loop, it can be divided into the following types
```mermaid
---
config:
  look: handDrawn
  theme: neutral
---
flowchart LR;
  8f8e6d117430("物品维护 Item maintenance")
  8f8e6d117430 --> d3zykgs7ac80("维护类型 Maintenance type")
  d3zykgs7ac80 --> d3zykhx2y880("时间维护物品 Time maintenance items")
  d3zykhx2y880 --> d3zyl5v5v7c0("固定循环保养周期 Fixed cycle maintenance cycle")
  d3zykhx2y880 --> d3zyl9k8t6w0("自定义保养周期 Custom maintenance cycle")
  d3zykgs7ac80 --> d3zykwpeis80("库存性维护物品 Inventory maintenance items")
  d3zykgs7ac80 --> d3zyl0140bk0("手动维护 Manual maintenance")

classDef default white-space: pre-line;
```

固定循环保养周期 Fixed cycle maintenance cycle
-
Exp: insurance, water purifier filter element, clean up household appliance dust
Items with a single cycle maintenance cycle

自定义保养周期 Custom maintenance cycle
-
In fact, it is possible to track various items, such as food and medicine, which expire at the end of their shelf life.
I can also track items with different cycle periods, such as my ID card (in my country, its validity period is 5 years, 10 years, 20 years, long-term, and each replacement will extend the validity period). I hope it can remind me to change my ID card

I have more imagination about the cycle
- It is necessary to add maintenance data after maintenance is completed, otherwise missed maintenance cycles will be accompanied by a permanent prompt
- Loose cycle: based on the first input time, not affected by the last maintenance time
- Rigorous cycle: After adding maintenance data, recalculate for the next cycle. If the maintenance time is delayed, the start time of the next cycle will also be postponed
- Use templates to manage Custom maintenance cycle
- Provide the option to keep items but no longer track or issue reminders

库存性维护物品 Inventory maintenance items
-
- Manage household inventory, such as tissues, garbage bags, and other items suitable for bulk purchases
- Considering that reverse management may be more convenient for some items, I believe it can be divided into two management directions: "greater than a certain value" and "less than a certain value"
- For some items, I think multiple judgment thresholds can be added
- Considering that some smart appliances such as smart water purifiers and smart door locks can display their remaining battery life, I think inventory can be recorded using floating-point numbers

Other brainstorming:
- After entering data more than twice, calculate the weighted average trend of the data and estimate the warning date
- Perhaps in the future, a certain degree of automatic maintenance can be achieved through APIs

手动维护 Manual maintenance
-
Keep a maintenance tracking mode similar to the current one and manually record maintenance time
Perhaps adding a scheduled date reminder feature to the existing mode

About Reminder
-
- In terms of time dimension, using days as the smallest unit, I believe it should be able to meet most scenarios.
- There may be many items that require prompt maintenance, and it should be possible to generate a list of all items based on reminder methods, cycle times, etc., which can be displayed on a new panel or sent to users through messaging channels on the same day

---
I also drew some imaginary display UI
![PixPin_2024-09-07_22-49-57-min](https://github.com/user-attachments/assets/5fb0d921-abe9-4837-a72c-e1bb7d60d4b7)

Complete mind map file
![物品维护 Item+maintenance-min](https://github.com/user-attachments/assets/e956eb53-0021-4b84-a2ad-b4e55d3cb2ba)

**Reply by KevinKTM** [reply]
Some good concepts here. Per my comment below, time based maintenance cycles should include hours (i.e. change oil on lawnmower every 30 hours).

**Comment by KevinKTM** [comment]
Second this request for maintenance logs - one essential ingredient (as with the inventory side) is the ability to add custom fields.  My use case for this software is a bit out in left field. I live on a sail boat - boat inventory is important because spare parts (and there are a lot of them) are stored wherever there is space so I often find I think I have a part when I don't (or vice versa) and when you do know you have it, where is it buried.  On the maintenance side, it would be for tracking maintenance of diesel engine, generator, electrical system, sails. Some items are on an as need basis (i.e. replace worn reefing line), some are time based (quarterly cut and polish) or engine oil (every 100 hours). 

With proper planning the use case for the software should become transparent. Your home may be a boat, an RV or a treehouse....

**Comment by tankerkiller125** [comment]
This is still on our roadmap, although probably further off than we originally thought.

There are some major things we'd like to get done and implemented in terms of overall project maintenance, and things like supporting multiple database types that really should be completed before implementing such a major feature (to prevent doubling the work to upgrade dependencies and swap things out).

Please do continue posting ideas on how this idea can be fully fleshed out and how it would work process wise though, along with how you would use the feature, having as much information as possible and as many ideas as possible can do nothing but help when we actually get to the point of implementation.

**Reply by KevinKTM** [reply]
I posted in another thread about home assistant integration. For me on a boat, I can watch the diesel engine hours and push that data into an HA sensor (vehicle mileage for land lubbers) and then use automations. Part of maintenance would be providing a maintenance schedule so I envision a card in HA that automatically dynamically updates with upcoming maintenance (the engine oil needs to be replaced in 14.2 hours (1524mi for the car). This could create an actionable notification that when the user acknowleges the oil change, the appropriate entry is made in the homebox maintenance log. The thing with HA is the sensor and automation infrastructure is already in place and its rich and powerful. Don't know if you use HA but take a look around - I haven't found anything that deals with maintenance - and its as much a part of home automation as anything else.  Maybe when you look a bit more you may steer the project more toward an HA integration?

Just found this, but ports of lots of apps into HA:
[https://github.com/alexbelgium/hassio-addons](url)

**Reply by Tomblarom** [reply]
@KevinKTM  great idea. I thought about using a small ESP32C3 to permanently read the mileage of my car and transmit it to homebox through HomeAssistant (ESPhome) over WiFi, so it's never necessary to actually manually enter any mileage values.

**Comment by schale01** [comment]
Being able to easily export the Maintenance record for any item would be very nice as well.

**Reply by jonhubby** [reply]
This proposal is well aligned with how homeowners typically manage upkeep, as linking maintenance history directly to each asset would make records far more useful than storing manuals and receipts alone. A structured approach that includes a defined “maintenance type” (e.g., service, inspection, part replacement), date, notes, cost, and optional attachments would cover most needs, while recurring schedules with reminders (such as quarterly or biannual tasks) would help maintain consistency. Referencing standard maintenance checklists, such as those outlined at https://www.mesa-plumbing.com/, can also help define what should be logged for systems like plumbing or water heaters. Additionally, the ability to filter and export maintenance logs per item would support warranty claims, resale documentation, and general record-keeping.

**Comment by Tomblarom** [comment]
I constantly put off regular maintenance tasks or forget they’re due. I want to stop that and Homebox feels like the perfect base for it. Maintenance belongs to the item you already track.

Right now Homebox is "inventory I update occasionally". With maintenance tracking, it becomes something users open regularly and it prevents real costs: premature wear, safety issues, missing service records (warranty/resale) and the classic “which bike/tool/jacket was serviced when?”. It also builds real insight over time by tracking part lifespans (e.g., how many km/hours a chain, brake pads, bearings, or filters actually lasted) so you can plan replacements, compare components, and spot unusually fast wear early.

## Some personal examples

### E-bikes
- Fork / rear shock service after **X riding hours** or **X km**
- Motor re-grease after **X km** (or hours)
- Bearings (pivot/headset/hub) after **X km** or **X years**
- Chain / chainring / cassette wear after **X km**
- Internal gear hub oil change after **X km** or **X months**

### Outdoor gear
- Gore-Tex DWR re-proofing every **X wears** (or when wetting-out), log product used
- Down jacket wash + re-loft every **X wears** / end of season
- Leather boots treatment (wax/cream/conditioner) every **X hikes** (or after wet/muddy trips) and at least every **X months**
- Sleeping bag wash every **X trips**; storage reminder (loft-preserving) every **X months**

### Cars
- Oil + filter every **X km** or **X months**
- Timing/aux belt every **X km** or **X years**
- Brakes: pads check every **X km**; rotors by **X pad sets** or thickness limit (log km + measurement)
- Brake fluid / coolant every **X years**
- Transmission/DSG/diff oil every **X km** or **X years**
- Spark plugs every **X km** (petrol) / glow plug checks (diesel)
- Air + cabin filter every **X km** or **X months**
- Tires: rotation/balance every **X km**; seasonal swap (summer↔winter) by **date**; replace by age after **X years**
- Washer fluid: switch to frost-safe before winter; refill check every **X weeks**
- Battery health check every **X months**; replace after **X years**
- Inspection / emissions due on a fixed calendar date
- A/C service (disinfect / leak check / recharge) every **X years**
- Rust protection / underbody wash after winter salt exposure (seasonal / yearly)

### Common household use cases
- HVAC / air purifier filter changes
- Smoke/CO detector test + battery replacement schedule
- Coffee machine descaling / cleaning cycles
- Tool / engine service hours (generator, mower, compressor, etc.)

## What Homebox needs
- Per item, a **Maintenance** section with:
  - Tasks with intervals in **time / km / hours / wears / cycles**
  - “Due after **X OR Y** (whichever comes first)” (e.g., 2 years or 2,000 km)
  - Service log entries (date, counter value, notes; optional cost/parts)
  - Dashboard view: **Due soon / Overdue** (notifications later)

I also looked at [LubeLogger](https://lubelogger.com/), but it’s car-focused and assumes only km/miles. Real-world home + hobby maintenance needs mixed units and mixed triggers. Adding this to Homebox would turn it into an actual maintenance system—not just a catalog—and I haven’t found a solid self-hosted alternative that handles these unit types cleanly.

**Comment by stephenlf** [comment]
I would also love this feature. Ideally, I would be able to scan a barcode to browse maintenance tasks and mark them complete (with document attachments for receipts, etc)
